packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewToolSectionsProvider.java

```diff
@@ -68,44 +105,188 @@ public List<ToolSection> handle(Object targetElement, Object diagramElement, Obj
+                    String sourceElementId = this.objectService.getId(viewTagetNodeDescriptionCandidates);
+
+                    Map<String, List<String>> parameters = this.urlParser.getParameterValues(diagramDescription.getId());
+                    List<String> values = Optional.ofNullable(parameters.get(IDiagramIdProvider.SOURCE_ID)).orElse(List.of());
+                    Optional<String> sourceId = values.stream().findAny();
+                    if (sourceId.isPresent()) {
+                        String formattedNodeDescriptionId = IDiagramIdProvider.NODE_DESCRIPTION_KIND + "?" + IDiagramIdProvider.SOURCE_KIND + "=view&" + IDiagramIdProvider.SOURCE_ID + "=" + sourceId.get() + "&" + IDiagramIdProvider.SOURCE_ELEMENT_ID + "=" + sourceElementId;
+                        Optional<NodeDescription> nodeDescriptionTargetCandidate = this.diagramDescriptionService.findNodeDescriptionById(diagramDescription, formattedNodeDescriptionId);
+                        if (nodeDescriptionTargetCandidate.isPresent()) {
+                            targetNodeDescriptionCandidates.add(nodeDescriptionTargetCandidate.get());
+                        }
+                    }
+                }
+            }
+            // @formatter:off
+            String toolId = this.idProvider.apply(edgeTool).toString();
+            var tool = SingleClickOnTwoDiagramElementsTool.newSingleClickOnTwoDiagramElementsTool(toolId)
+                    .label(edgeTool.getName())
+                    .imageURL(ViewToolImageProvider.EDGE_CREATION_TOOL_ICON)
+                    .candidates(List.of(SingleClickOnTwoDiagramElementsCandidate.newSingleClickOnTwoDiagramElementsCandidate()
+                            .sources(List.of(nodeDescription))
+                            .targets(targetNodeDescriptionCandidates)
+                            .build()))
+                    .handler(variableManager -> {
+                        return null;
+                    })
+                    .build();
+            // @formatter:on
+            tools.add(tool);
+        }
+        return tools;
     }
 
-    private List<ToolSection> getEdgeToolSections(DiagramDescription diagramDescription, EdgeDescription edgeDescription) {
+    private List<ITool> createEdgePaletteTools(org.eclipse.sirius.components.view.EdgeDescription viewEdgeDescription) {
+        List<ITool> tools = new ArrayList<>();
+
+        List<NodeTool> paletteSingleTargetTools = new ToolFinder().findNodeTools(viewEdgeDescription);
+        for (NodeTool nodeTool : paletteSingleTargetTools) {
+            // @formatter:off
+            String toolId = this.idProvider.apply(nodeTool).toString();
+            var tool = SingleClickOnDiagramElementTool.newSingleClickOnDiagramElementTool(toolId)
+                    .label(nodeTool.getName())
+                    .imageURL(ViewToolImageProvider.NODE_CREATION_TOOL_ICON)
+                    .handler(variableManager -> {
+                        return null;
```

Comment:
Add a failure handler with the unexpected error message.
Log an error in the console as well